src/test/java/com/example/user_registration/repository/ContactDetailsRepositoryTest.java

```diff
@@ -0,0 +1,56 @@
+package com.example.user_registration.repository;
+
+import com.example.user_registration.entity.ContactDetails;
+import com.example.user_registration.repository.ContactDetailsRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@DataJpaTest
+public class ContactDetailsRepositoryTest {
+
+    @Mock
+    private ContactDetailsRepository contactDetailsRepository;
```

Comment:
You made the same mistake again. Like i said, since it is a repository test we should not mock the repository and let it connect to the database. you can mock the repository for a service test